### DISCUSSION TITLE
onUploadProgress of Axios not working in AlpineJS Data

### DISCUSSION THREAD

**Discussion by rexwebmedia** [alpinejs/alpine#3882] [7. Bugs (archived)]
The onUploadProgress is working when using it normally with vanilla js, but the same code doesn't return the uploaded progress event data when called from Alpine JS.

Here's the minimal HTML and JS code. The first input works normally and the progress data is shown in console, but second input does not work.

Thanks.

* <https://cdnjs.cloudflare.com/ajax/libs/axios/1.6.2/axios.min.js>
* <https://cdnjs.cloudflare.com/ajax/libs/alpinejs/3.13.3/cdn.js>

```html
<!-- check console after uploading files -->
Vanilla JS
<input type="file" id="normalFileInput" multiple="multiple" />
<hr>

<div x-data="assetsDropzoneData">
  AlpineJS
  <input type="file" multiple="multiple" @change="handleAssetsChange($event)" />
</div>
```

```js
var route = "https://v2.convertapi.com/upload";

(function () {
  var normalFileInput = document.getElementById("normalFileInput");

  normalFileInput.addEventListener("change", function (e) {
    var files = e.target.files;
    if (files && files.length) {
      var data = new FormData();
      for (var i = 0; i < files.length; i++) {
        data.append("assets[]", files[i]);
      }
      axios
        .post(route, data, {
          onUploadProgress: function (ev) {
            console.log("Uploading... ", ev.loaded, ev.total);
          }
        })
        .then(function (res) {
          console.log(res);
        })
        .catch(function (err) {
          console.log(err);
        });
    }
  });
})();

function assetsDropzoneData() {
  return {
    handleAssetsChange(e) {
      var self = this;
      var files = e.target.files;
      if (files && files.length) {
        var data = new FormData();
        for (var i = 0; i < files.length; i++) {
          data.append("assets[]", files[i]);
        }
        axios
          .post(route, data, {
            onUploadProgress: function (ev) {
              console.log("Uploading... ", ev.loaded, ev.total);
            }
          })
          .then(function (res) {
            console.log(res);
          })
          .catch(function (err) {
            console.log(err);
          });
      }
    }
  };
}
function alpine_init() {
  Alpine.data("assetsDropzoneData", assetsDropzoneData);
}
document.addEventListener("alpine:init", alpine_init);
```

**Comment by ekwoka** [comment]
It works fine.

If you can replicate it in codepen or something, that will be better. But this works as is.

**Reply by rexwebmedia** [reply]
I tried to create a codepen, but Alpine needs to be loaded with "defer", so that's not working.

https://codepen.io/rexarvind/pen/bGzKGXB

**Reply by ekwoka** [reply]
just import it from `https://esm.sh/alpinejs` and call Alpine.start()

works just fine.

**Reply by rexwebmedia** [reply]
It is still not loading Alpine, and that upload progress event doesn't work even after using it locally.

![image](https://github.com/alpinejs/alpine/assets/63487995/c091f7dc-c420-49d3-96fa-0c9f06a1db72)

**Reply by ekwoka** [reply]
You need to call start after registering data.

**Comment by SimoTod** [comment]
You need to call start after you set up the listener for alpine:init
otherwise the event gets fired before your code is evaluated and the
listener will never run. Same if you use the cdn (the cdn script should be
deferred and if both scripts are deferred, Alpine cdn must go after).
